### PR TITLE
[fix]: [CDS-49822]: add correct commitId

### PR DESCRIPTION
**Pull request by tmacari-harness**
## Describe your changes
[fix]: [CDS-49822]: add correct commitId
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-49822]: https://harness.atlassian.net/browse/CDS-49822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42445)

### CODE REVIEW

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformApplyTaskHandlerTest.java [hunk 1]
```diff
@@ -147,7 +144,6 @@ public void testApplyWithArtifactoryConfigAndVarFiles() throws IOException, Time
     assertThat(response.getCommandExecutionStatus()).isEqualTo(CommandExecutionStatus.SUCCESS);
     Files.deleteIfExists(Paths.get(outputFile.getPath()));
     Files.deleteIfExists(Paths.get("sourceDir"));
-    verify(terraformBaseHelper, times(1)).addVarFilesCommitIdsToMap(any(), any(), any());
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformApplyTaskHandlerTest.java:150]: minor: from my understanding we need to update test naming, as it doesn't check varFiles here.

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformDestroyTaskHandlerTest.java [hunk 2]
```diff
@@ -137,7 +134,6 @@ public void testApplyWithArtifactoryConfigAndVarFiles() throws IOException, Time
     assertThat(response.getCommandExecutionStatus()).isEqualTo(CommandExecutionStatus.SUCCESS);
     Files.deleteIfExists(Paths.get(outputFile.getPath()));
     Files.deleteIfExists(Paths.get("sourceDir"));
-    verify(terraformBaseHelper, times(1)).addVarFilesCommitIdsToMap(any(), any(), any());
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformDestroyTaskHandlerTest.java:140]: minor: from my understanding we need to update test naming, as it doesn't check varFiles here.

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformDestroyTaskHandlerTest.java [hunk 3]
```diff
@@ -163,7 +159,6 @@ public void testDestoryWithArtifactoryConfigAndBackendConfig()
     assertThat(response.getCommandExecutionStatus()).isEqualTo(CommandExecutionStatus.SUCCESS);
     Files.deleteIfExists(Paths.get(outputFile.getPath()));
     Files.deleteIfExists(Paths.get("sourceDir"));
-    verify(terraformBaseHelper, times(1)).addBackendFileCommitIdsToMap(any(), any(), any());
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformDestroyTaskHandlerTest.java:166]: minor: same seems here we don't check backendFile commt but we create a backend-config file

**Review by abosii-harness — APPROVED** [review]
> Reviewed 9 of 10 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42445)* status: 10 file(s) to review. 1 of 2 approvals obtained.

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformDestroyTaskHandlerTest.java (hunk 3 above)
Comment (tmacari-harness) [reply 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformDestroyTaskHandlerTest.java:166]: we do not check for addBackendFileCommitIdsToMap() anymore, but backendConfigFiles are used in this test, TerraformTaskNGParameters contains them.

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformApplyTaskHandlerTest.java (hunk 1 above)
Comment (tmacari-harness) [reply 930-delegate-tasks/src/test/java/io/harness/delegate/task/terraform/handlers/TerraformApplyTaskHandlerTest.java:150]: we just do not check if terraformBaseHelper.addVarFilesCommitIdsToMap().
The test itself uses var files.
getTerraformTaskParametersWithArtifactoryConfig() returns a TerraformTaskNGParameters that contains var files.

**Review by vlica-harness — APPROVED** [review]
> Reviewed 9 of 10 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42445)* status: :shipit: complete! ✓

**Review by vlica-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42445)* status: :shipit: complete! ✓

**Review by vlica-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42445)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by alebuh — APPROVED** [review]
> Reviewed 8 of 10 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42445)* status: :shipit: complete! ✓

**Review by arvind-choudhary-h — COMMENTED** [review]
> Reviewed 8 of 10 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42445)* status: :shipit: complete! ✓

**Pull request merged**